src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -205,6 +206,15 @@ protected RequestConfig buildRequestConfig() {
             .build();
   }
 
+  /**
+   * Sub-classes can override specific behaviour of {@link org.apache.http.config.SocketConfig}.
+   */
+  protected SocketConfig buildSocketConfig() {
+    return SocketConfig.custom()
```

Comment:
I think this can return null if there is no timeout.  From what I see, a null SocketConfig is supported and what occurs normally (if you have no customizations).

Reply:
Sorry I missed your comment :( -> DONE